code/modules/mob/living/simple_animal/bot/hygienebot.dm

```diff
@@ -0,0 +1,356 @@
+	frustration = 0
+	mode = BOT_HUNT
+	stop_washing()
+	INVOKE_ASYNC(src, .proc/handle_automated_action)
+
+/mob/living/simple_animal/bot/hygienebot/proc/look_for_lowhygiene()
+	for (var/mob/living/carbon/human/H in view(7,src)) //Find the NEET
+		if((H.name == oldtarget_name) && (world.time < last_found + 100))
+			continue
+		if(!check_purity(H)) //Theyre impure
+			target = H
+			oldtarget_name = H.name
+			speak("Unhygienic client found, Please stand still so I can clean you")
+			playsound(loc, 'sound/effects/hygienebot_happy.ogg', 60, 1)
+			visible_message("<b>[src]</b> points at [H.name]!")
+			mode = BOT_HUNT
+			INVOKE_ASYNC(src, .proc/handle_automated_action)
+			break
+		else
+			continue
+
+/mob/living/simple_animal/bot/hygienebot/proc/start_washing()
+	washing = TRUE
+	update_icon()
+
+/mob/living/simple_animal/bot/hygienebot/proc/stop_washing()
+	washing = FALSE
+	update_icon()
+
+
+
+/mob/living/simple_animal/bot/hygienebot/get_controls(mob/user)
+	var/dat
+	dat += hack(user)
+	dat += showpai(user)
+	dat += text({"
+<TT><B>Hygienebot X2 controls</B></TT><BR><BR>
+Status: []<BR>
+Behaviour controls are [locked ? "locked" : "unlocked"]<BR>
+Maintenance panel panel is [open ? "opened" : "closed"]"},
+
+"<A href='?src=[REF(src)];power=[TRUE]'>[on ? "On" : "Off"]</A>" )
+
+	if(!locked || issilicon(user) || IsAdminGhost(user))
+		dat += text({"<BR> Auto Patrol: []"},
+
+"<A href='?src=[REF(src)];operation=patrol'>[auto_patrol ? "On" : "Off"]</A>" )
+	return	dat
+
+/mob/living/simple_animal/bot/hygienebot/proc/check_purity(mob/living/L)
+	if(emagged && L.stat != DEAD)
+		return FALSE
+
+	var/obj/item/head = L.get_item_by_slot(ITEM_SLOT_HEAD)
+	if(head)
+		if(HAS_BLOOD_DNA(head))
+			return FALSE
+
+	var/obj/item/mask = L.get_item_by_slot(ITEM_SLOT_MASK)
+	if(mask)
+		if(HAS_BLOOD_DNA(mask))
+			return FALSE
+
+	var/obj/item/uniform = L.get_item_by_slot(ITEM_SLOT_ICLOTHING)
+	if(uniform)
+		if(HAS_BLOOD_DNA(uniform))
+			return FALSE
+
+	var/obj/item/suit = L.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+	if(suit)
+		if(HAS_BLOOD_DNA(suit))
+			return FALSE
+
+	var/obj/item/feet = L.get_item_by_slot(ITEM_SLOT_FEET)
+	if(feet)
+		if(HAS_BLOOD_DNA(feet))
+			return FALSE
+	return TRUE
+
+/mob/living/simple_animal/bot/hygienebot/proc/wash_atom(atom/A)
+	SEND_SIGNAL(A, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_WEAK)
+	if(emagged)
+		A.fire_act()
+		return //lol pranked no cleaning besides that
+	if(isobj(A))
+		wash_obj(A)
+	else if(isturf(A))
+		wash_turf(A)
+	else if(isliving(A))
+		wash_mob(A)
+
+	contamination_cleanse(A)
+
+/mob/living/simple_animal/bot/hygienebot/proc/wash_obj(obj/O)
+	. = SEND_SIGNAL(O, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_WEAK)
+	O.remove_atom_colour(WASHABLE_COLOUR_PRIORITY)
+
+
+/mob/living/simple_animal/bot/hygienebot/proc/wash_turf(turf/tile)
+	SEND_SIGNAL(tile, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_WEAK)
+	tile.remove_atom_colour(WASHABLE_COLOUR_PRIORITY)
+	for(var/obj/effect/E in tile)
+		if(is_cleanable(E))
+			qdel(E)
```

Comment:
This is handled by wash_obj.

Reply:
this is mostly all copypasta'd from `/obj/machinery/shower/proc/wash_turf(turf/tile)`